AutomatticTracks/src/test/java/com/automattic/android/tracks/CrashLogging/CrashLoggingTest.kt

```diff
@@ -0,0 +1,78 @@
+package com.automattic.android.tracks.CrashLogging
+
+import com.automattic.android.tracks.TracksUser
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito.*
+import org.mockito.junit.MockitoJUnit
+import org.mockito.junit.MockitoRule
+import org.mockito.quality.Strictness
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.RuntimeEnvironment
+
+@RunWith(RobolectricTestRunner::class)
+class CrashLoggingTest {
+
+    @get:Rule
+    var rule: MockitoRule = MockitoJUnit.rule().strictness(Strictness.STRICT_STUBS)
+
+    lateinit var dataProvider: CrashLoggingDataProvider
+    var user: TracksUser? = null
+
+    val testUser = TracksUser(
+            "testUserId",
+            "testEmail",
+            "testUsername"
+    )
+
+    private fun initialize(currentUser: TracksUser? = null) {
+
+        user = if (currentUser != null) spy(currentUser) else null
+
+        dataProvider = spy(FakeDataProvider(currentUser = user))
+
+        CrashLogging.start(
+                context = RuntimeEnvironment.systemContext,
+                dataProvider = dataProvider
+        )
+    }
+
+    @Test(expected = UnsupportedOperationException::class)
+    fun shouldThrowExceptionOnRequest() {
+        CrashLogging.crash()
+    }
+
+    @Test
+    fun `should call for required arguments during initialisation`() {
+        initialize()
+
+        verify(dataProvider, times(1)).sentryDSN()
```

Comment:
I didn't want to use `PowerMock` to mock static `Sentry` or `SentryAndroid` so I've decided to spy on our `CrashLoggingDataProvider` to make sure methods with data are requested (so we can assume that are assigned).